tools/doc/addon-verify.js

```diff
@@ -1,68 +1,52 @@
 'use strict';
 
-const fs = require('fs');
-const path = require('path');
-const marked = require('marked');
+const { mkdir, readFileSync, writeFile } = require('fs');
+const { resolve } = require('path');
+const { lexer } = require('marked');
 
-const rootDir = path.resolve(__dirname, '..', '..');
-const doc = path.resolve(rootDir, 'doc', 'api', 'addons.md');
-const verifyDir = path.resolve(rootDir, 'test', 'addons');
+const rootDir = resolve(__dirname, '..', '..');
+const doc = resolve(rootDir, 'doc', 'api', 'addons.md');
+const verifyDir = resolve(rootDir, 'test', 'addons');
 
-const contents = fs.readFileSync(doc).toString();
-
-const tokens = marked.lexer(contents);
+const tokens = lexer(readFileSync(doc, 'utf8'));
+const addons = {};
 let id = 0;
-
 let currentHeader;
-const addons = {};
-tokens.forEach((token) => {
-  if (token.type === 'heading' && token.text) {
-    currentHeader = token.text;
-    addons[currentHeader] = {
-      files: {}
-    };
+
+const validNames = /^\/\/\s+(.*\.(?:cc|h|js))[\r\n]/;
+tokens.forEach(({ type, text }) => {
+  if (type === 'heading') {
```

Comment:
Is the check for `text` required here?

Reply:
If I tested correctly, there cannot be an empty heading: they are not recognized as headings by `marked`:
```js
const marked = require('marked ');

const md = `
# heading

text

#

text
`;

console.log(marked.lexer(md));
```
```js
[ { type: 'heading', depth: 1, text: 'heading' },
  { type: 'paragraph', text: 'text' },
  { type: 'paragraph', text: '#' },
  { type: 'paragraph', text: 'text' },
  links: {} ]
```